nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -395,6 +399,29 @@ def simulator_run_main(self):
             run_status = 1
         return run_status
 
+    def split_gpus(self, gpus) -> ([str], bool):
+        gpus = gpus.replace(" ", "")
+        if not (gpus.startswith("[") and gpus.endswith("]")):
+            return None, False
+        result = []
+        index = 0
+        while index < len(gpus):
+            if gpus[index] != "[":
+                return None, False
+            a = ""
+            while True:
+                index += 1
+                if gpus[index] == "]":
+                    break
+                a += gpus[index]
+            result.append(a)
+            index += 1
+            if index < len(gpus) and gpus[index] != ",":
+                return None, False
+            else:
+                index += 1
+        return result, True
+
```

Comment:
Suggest move this function outside of the class, since it does not need anything from class.

Reply:
I am not quite sure about this. This is a unique function only used by this class.

Reply:
Moved the split_gpus() to fed_utils.